### ISSUE TITLE
Shipping this to users

### ISSUE THREAD

**Issue by stefan-reich** [javafxports/openjdk-jfx#285]
How can I ship JavaFX to users? So far I only need them to have Java installed which is practically a given. Thus my application is a simple jar file.

If I require JavaFX and want to support Java 11+, can I still get away with a single jar file?

**Comment by Tristan971** [comment]
Yes, you just use the maven/gradle dependencies bundled in a uber-jar like any other dependency for end-user jars :) They are here : https://search.maven.org/search?q=g:org.openjfx
More info on https://openjfx.io/

**Comment by Tristan971** [comment]
Yes. I'm not sure if someone did it specifically for JavaFX but you can go with maven's shade plugin for fat-jar generation that auto-bundles every needed lib into the jar https://maven.apache.org/plugins/maven-shade-plugin/examples/executable-jar.html

**Comment by kevinrushforth** [comment]
If you mean to include all of the jars for all platforms in an attempt to have a single binary that can run on all platforms, I very much doubt that this will work (at least not without some very clever, custom unpacking code). @johanvos or @jperedadnr might have some suggestions for you.

**Comment by nlisker** [comment]
Have a look at https://openjfx.io/openjfx-docs/index.html#fat-jar. There are 3 modules that have a native version, so if you use any of them you'll need to include the 3 versions of them. Iv'e done it and the same jar runs on Windows and Mac (didn't test Linux yet).
